yare-engine/src/main/java/com/sabre/oss/yare/engine/executor/DefaultRulesExecutor.java

```diff
@@ -129,8 +134,10 @@ public boolean clear() {
     }
 
     private void evaluateSequentially(RuntimeRules runtimeRules, Object result, Map<String, Object> factMap) {
-        for (RuntimeRules.ExecutableRule executableRule : runtimeRules.getExecutableRules()) {
-            PredicateContext context = new PredicateContext(executableRule.getRuleId(), result, factMap, executableRule.getAttributes());
+        List<RuntimeRules.ExecutableRule> executableRules = runtimeRules.getExecutableRules();
+        for (int i = 0; i < executableRules.size() && !stopProcessingListener.isEvaluationTerminated(); ++i) {
```

Comment:
Since this condition is pretty long, extracting it to separate helper method would be more readable.

Reply:
I think good idea, but I don't know what name would be appropriate for the condition.

Reply:
I've tried 'while' loop and 'for' loop with `continueIterate()` condition, but I think it is not more readable.  Any ideas? :)